File: servertier/src/main/scala/sgad/servertier/dataaccess/data/shareddata/Resource.scala
/**
 * FILE: Resource.scala
 * PERCORSO /Codice/sgad/servertier/src/main/scala/sgad/servertier/dataaccess/data/shareddata
 * DATA CREAZIONE: 18 Febbraio 2014
 * AUTORE: ProTech
 * EMAIL: protech.unipd@gmail.com
 *
 * Questo file è proprietà del gruppo ProTech, viene rilasciato sotto licenza Apache v2.
 *
 * DIARIO DELLE MODIFICHE:
 * 2014-02-18 - Creazione della classe - Segantin Fabio
 */
package sgad.servertier.dataaccess.data.shareddata

/**
 * Classe per la memorizzazione delle informazioni su una risorsa.
 * @constructor
 * @param resourceName Il nome della risorsa.
 */
class Resource(private val resourceName : String) {
	if(resourceName == null)
		throw new IllegalArgumentException
	/**
	 * Metodo getter per il nome risorsa.
	 * @return Il nome della risorsa.
	 */
	def getResourceName = resourceName
	/**
	 * Fornisce la chiave identificativa della risorsa.
	 * @return Il nome della risorsa.
	 */
	def getKey = getResourceName
	/**
	 * Verifica se due risorse sono uguali.
	 * @param risorsa La risorsa con cui verificare l'uguaglianza.
	 * @return Vero o falso in base all'esito del confronto.
	 */
	override def equals(risorsa : Any):Boolean = {
		if(!risorsa.isInstanceOf[Resource])
			false
		else
			this.resourceName == risorsa.asInstanceOf[Resource].resourceName
	}
}

File: clienttier/model/generaldata/Resource.js
/**
 * FILE: Resource.js
 * PERCORSO /Codice/sgad/clienttier/model/generaldata/Resource.js
 * DATA CREAZIONE: 12 Febbraio 2014
 * AUTORE: ProTech
 * EMAIL: protech.unipd@gmail.com
 *
 * Questo file è proprietà del gruppo ProTech, viene rilasciato sotto licenza Apache v2.
 *
 * DIARIO DELLE MODIFICHE:
 * 2014-02-12 - Creazione della classe - Gatto Francesco
 */
 
 /**
 * Classe che rappresenta una risorsa.
  * @constructor
  * @param {string} resourceName Il nome della risorsa.
 */
function Resource(resourceName) {
	/**
	 * Il nome della risorsa.
	 * @type {string}
	 * @private
	 */
	this.resourceName = resourceName;

}

/**
 * Controlla l'accesso al nome della risorsa.
 * @return {string} Nome della risorsa.
 */
Resource.prototype.getResourceName = function() {
	return this.resourceName;
};

/**
 * Fornisce la chiave che identifica la risorsa.
 * @returns {string} La chiave che identifica la risorsa.
 */
Resource.prototype.getKey = function() {
    return this.getResourceName();
};

/**
 * Ridefinizione del metodo valueOf di Resource.
 * @override
 * @returns {string} L'identificativo della risorsa.
 */
Resource.prototype.valueOf = function() {
    return JSON.stringify({resourceName : this.getKey()});
};